[i18n] Integrate 8.12.0 Translations

Bamieh opened the pull request:
Integrating latest translations extracted from main branch.
Skipping backports from main to target branches since the `i18n_check` might trim unused translations that are still used in different branches. Integration script is ran against each target branch separately.

Bamieh committed:
Integrate zh-CN translations from 8.12.0 extraction

Bamieh committed:
Integrate ja-JP translations from 8.12.0 extraction

Bamieh committed:
Integrate fr-FR translations from 8.12.0 extraction

Bamieh commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into i18n/sync_8.12.0_translations_to_main

Bamieh commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into i18n/sync_8.12.0_translations_to_main

Bamieh commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into i18n/sync_8.12.0_translations_to_main

TinaHeiligers commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into i18n/sync_8.12.0_translations_to_main

TinaHeiligers commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into i18n/sync_8.12.0_translations_to_main

pgayvallet commented:
From the test failure:

```
debg Find.allByCssSelector('.euiFilterSelectItem[title="求和"]') with timeout=2500
warn comboBox.setElement - Could not find option [求和], using first
```

Look like the label of an option that was used by some FTR test was changed by the translation, so the test probably needs to be updated accordingly.

tonyghiani committed:
[Logs Shared] Types refactor (#173819)

## 📓 Summary

Closes #173493 

This work improves typing for the `dynamic` util for lazily exported
components, catching a couple of typing issues.

---------

Co-authored-by: Marco Antonio Ghiani <marcoantonio.ghiani@elastic.co>

machadoum committed:
[Security Solution] Refactor SeverityFilterGroup to use EuiSelectable instead of EuiFilterSelectItem (#174912)

`EuiFilterSelectItem` was deprecated by the EUI team

Flaky test runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/4879

## Summary

### Before
<img width="200"
src="https://github.com/elastic/kibana/assets/1490444/233ff8c5-09c4-4afa-a719-1e0cae8f0396"/>
<img width="213"
src="https://github.com/elastic/kibana/assets/1490444/8cc262d0-047a-4ef7-b182-ab42df6923f9"/>

### After
<img width="200"
src="https://github.com/elastic/kibana/assets/1490444/d5a822c5-1d52-4880-81c7-83ab82df01e3"/>
<img width="210"
src="https://github.com/elastic/kibana/assets/1490444/6ce840c0-18d5-42fa-b4ab-734478b023a8"/>


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))

lcawl committed:
[DOCS] Refresh cases table screenshot (#174871)

dominiqueclarke committed:
[SLO] api integration tests stateful (#173236)

## Summary

Adds tests for basic SLO api routes, including:
1. Find slos
2. Get slo by id
3. Get slo definitions
4. Get slo instances
5. Create slo
6. Delete slo
7. Update slo
8. Reset slo

The create slo tests include some basic assertions that the resulting
calculated SLO is correct.

These tests do not cover:
1. SLOs in spaces
2. SLO permissions model

Passed flaky test runner for 200 iterations:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/4595#_

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: shahzad31 <shahzad31comp@gmail.com>
Co-authored-by: Kevin Delemme <kdelemme@gmail.com>

drewdaemon committed:
[Lens][Docs] add metric coloring doc (#174405)

## Summary

Fix https://github.com/elastic/kibana/issues/174379

Adds an FAQ section for the coloring behavior of the metric
visualization

<img width="847" alt="Screenshot 2024-01-08 at 12 51 27 PM"
src="https://github.com/elastic/kibana/assets/315764/79a74a79-af07-474d-aceb-124d62dbee76">

---------

Co-authored-by: amyjtechwriter <61687663+amyjtechwriter@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

crespocarlos committed:
[APM] Fix error count threshold alert (#174907)

closes https://github.com/elastic/kibana/issues/167974

## Summary

Follows a similar approach to
https://github.com/elastic/kibana/pull/174327, to prevent the tests from
running without the necessary synthtrace data.

https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/4878

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Bamieh committed:
i18n fix

mistic committed:
skip flaky suite (#174066)

XavierM committed:
[RAM] Stack management rules available with only ml feature (#174791)

## Summary

Fix => https://github.com/elastic/kibana/issues/173573

mistic committed:
skip flaky suite (#174819)

mistic committed:
skip flaky suite (#174259)

amyjtechwriter committed:
[DOCS] Adds more information how to handle a breaking change in 8.10.0 - SLOs (#174901)

A breaking change was introduced in 8.10.0, which makes SLOs created
before migration not work. They need to be removed and reset-up. The
release note in 8.10.0 did not make this clear. More information has
been added to the release note in this PR including steps to remove the
SLOs that no longer work.

Closes: #173938

cnasikas committed:
[Cases] Fix description flaky tests (#174903)

## Summary

Fixes: https://github.com/elastic/kibana/issues/174323,
https://github.com/elastic/kibana/issues/174322,
https://github.com/elastic/kibana/issues/174321,
https://github.com/elastic/kibana/issues/164049,
https://github.com/elastic/kibana/issues/164048,
https://github.com/elastic/kibana/issues/164045

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### For maintainers

- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

devcorpio committed:
[Docs] remove .map from bundle_filepath value (#174834)

Part of
https://github.com/elastic/observability-docs/issues/3354#issuecomment-1849730927

## Summary

`bundle_filepath` should not contain `.map`.

drewdaemon committed:
[Docs] Remove broken weak links fix plan (#174932)

## Summary

The broken weak links problem wasn't fixed in 8.1, so this statement is
now confusing.

An aliasing workaround [was
added](https://github.com/elastic/kibana/pull/149021) in 8.8 for those
customers who need it.

Better solutions
- use the [new links
panel](https://www.elastic.co/guide/en/kibana/current/dashboard-links.html)
for navigation between dashboards
- use [shareable
dashboards](https://github.com/elastic/kibana/issues/167901) to make
analytics stuff available in multiple spaces instead of copying (not yet
available)

Bamieh committed:
debug line

TattdCodeMonkey committed:
[Console][Serverless Search] Introduce Embeddable Console, Show Embeddable Console in Serverless Search (#174643)

## Summary

This PR introduces the `renderEmbeddableConsole` from the console
plugin. This allows other plugin's to embed a portable version of the
dev console on their own pages. Initially the embeddable console is
being added to Serverless Search getting started & connectors pages.
There will be follow-up work to add the embeddable console to other
pages for serverless search. Then again work to add the remote console
to Search and other pages in v8.13.0 of Kibana.

### Screenshots
Closed

![image](https://github.com/elastic/kibana/assets/1972968/8bf69034-ae79-41f7-9161-4baf2b9e7a17)
Open

![image](https://github.com/elastic/kibana/assets/1972968/8fac53c1-997f-4009-b2c2-c085cc8ee9ec)

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

christineweng committed:
[Security Solution] Alert flyout - update document id in analyzer preview and same ancestry (#174651)

## Summary

Address: https://github.com/elastic/kibana/issues/169373

This PR updates the use of `kibana.alert.ancestor.id` to `_id`
(available in flyout context as `eventId`) in analyzer preview and
alerts by ancestry. This change allows upgrade from 7.x kibana to 8.10+
to utilize analyzer preview.

No UI change introduced.

**How to test**
- Analyzer preview should match that of prior to the change
- Alert by ancestry in correlations overview (right section) and
correlations tab (left section -> Insights) should match that of prior
to the change
- Analyzer preview should match the analyzer viewer in alerts table


### Checklist
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

tsullivan committed:
[Reporting] Track retryAt time on the task instance (#174409)

## Summary

Initial part of https://github.com/elastic/kibana/issues/131852

This PR moves towards auto-calculating the maximum timeouts calculated
based on the timing context provided by the running task instance.

### Other changes
* Added an optional logger parameter to the `getScreenshots` function.
When a logger is provided, the logs created by the screenshot plugin
will have the contextual tags added by the calling code.
  * Before
<img width="1198" alt="image"
src="https://github.com/elastic/kibana/assets/908371/f68a102e-6af2-4863-aedb-52f1e4a099d8">
  * After
<img width="1200" alt="image"
src="https://github.com/elastic/kibana/assets/908371/2dd4c947-ffa6-4cb3-b8a2-22893f49ddb7">

* Fixed an unreported bug where browser timezone was not utilized in PNG
reports.

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Bamieh committed:
remove debug line

jaredburgettelastic committed:
Risk score asset criticality upgrade (#174474)

## Summary

This PR implements the decision made in [this Architecture Decision
Record (ADR)](https://github.com/elastic/entity-analytics/pull/5/files)
to change the `dynamic` property of the risk score latest index to
`false`. This change is now performed:
- Any time the risk engine is enabled (meaning `false` is now the
default value)
- Any time risk scores are persisted (specifically, we upgrade
**before** persisting said scores)

See that ADR for additional details and context.

## Testing

This PR includes unit tests to ensure that the basic upgrade workflow is
conducted when persisting risk scores.

Additionally, FTR tests are also included that simulates the upgrade
workflow, ensuring that mappings are unchanged afterward, with the
exception of the `dynamic` property.

Flaky test runner results:
https://buildkite.com/elastic/kibana-pull-request/builds/186924

### Checklist


- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

delanni committed:
[Ops] Display plugin version diff (#174625)

## Summary
Closes: https://github.com/elastic/kibana/issues/173347

Although the info could be gathered in the current state, it's
inconvenient to read through the JSON file and compare from/to versions,
this PR adds direct display of a critical indicator on the SO migration
changes - the version change.

This is retrieved from the `modelVersions` and displayed in the CLI
output as well as the release assistant job's UI.

The CLI now has changed a bit, there's a `--json` flag option to ask
explicitly for the bloated JSON output (that's often only parsed by code
anyway) - by default it's only human readable logs.

Looks like this in the CLI
```
info Snapshot already exists at: /var/folders/47/4x67d0_n7hv_fmg47slbqj0h0000gn/T/7470d2136dec44ac45d5db091a5ccf873b00adaf.json
 info Snapshot already exists at: /var/folders/47/4x67d0_n7hv_fmg47slbqj0h0000gn/T/f8e992393a45260b4001a97cac3e6d154e57ae82.json
 info Snapshots compared: 7470d2136dec <=> f8e992393a45. Changed: alert, ingest-outputs, epm-packages
 info ✅ alert: 1 => 0
 info 🚨 ingest-outputs: 3 => 1
 info ✅ epm-packages: 1 => 0
 info Output written to: xoxo.json
```

Looks like this in the release assistant: 
<img width="539" alt="Screenshot 2024-01-12 at 11 24 52"
src="https://github.com/elastic/kibana/assets/4738868/36f80b20-4cf3-4b7f-8277-efa16ed16c90">

jbcrail committed:
[Profiling] Remove unused fields from stackframe metadata (#171151)

## Summary

This PR removes unused legacy fields that originated from a previous
iteration of Universal Profiling.

The `StackFrameMetadata` interface will soon be replicated in the
Universal Profiling plugin in Elasticsearch, so we are preemptively
removing these fields to minimize future work. It also will have a side
effect of reducing the payload size from Elasticsearch down to the UI.

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

### For maintainers

- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Cauê Marcondes <55978943+cauemarcondes@users.noreply.github.com>

kibanamachine committed:
[api-docs] 2024-01-17 Daily api_docs build (#174983)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/585

akanshi-elastic committed:
Add new RCF for GCS and ABS in native connector (#174512)

Summary
[Related to
https://github.com/elastic/connectors/issues/1819](https://github.com/elastic/connectors/issues/1831)

1. Add new config parameter in ABS Connector `containers`
2. Add new config parameter in GCS Connector `buckets`
3. please add label 8.13 for this PR

Checklist
Delete any items that are not applicable to this PR.

- [X] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)

For maintainers
 

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Dosant committed:
[ES|QL] CSV Reporting in Discover (#174511)

## Summary

close https://github.com/elastic/kibana/issues/173390
This PR enables CSV report generation with ES|QL in Discover. 

Pre this PR, there are two report types for generating CSV reports with
Discover:
-
https://github.com/elastic/kibana/blob/main/packages/kbn-reporting/export_types/csv/csv_searchsource.ts
- old deprecated report type that relies on
`SerializedSearchSourceFields`. This type is still currently used by
Discover UI and we plan to migrate away from it
https://github.com/elastic/kibana/issues/151190
-
https://github.com/elastic/kibana/blob/main/packages/kbn-reporting/export_types/csv/csv_v2.ts
- new report type that relies on Discover locator. Currently, it can
only generate reports from searches backed by saved search and this
report type was implemented as a public-facing API for simple report
generation outside Kibana.

Since we plan to migrate to the v2 report type and search source is not
needed for es|ql, this PR implements es|ql csv reporting based on the v2
report type.

Initially, I wanted to create new new report type similar to v2 just for
es|ql, but it turned out to be a lot more boilerplate code without a
significant benefit. you can see it in the PR
https://github.com/elastic/kibana/pull/174448. So I changed my mind and
the current PR adds es|ql capabilities inside the existing csv_v2
report. This is convenient as the input is the same (Discover locator),
the output is the same csv file and meta information, and telemetry is
also the same.

As of this PR, the ES|QL report is capable of: 
- Using es|ql query from the locator 
- Adding time range filter if present in the locator. time field is
picked from the data view (it is available in the locator, but otherwise
is not used for es|ql). Other filters are also passed if available in
the locator
- Keeps into account "columns" from the locator.
- Similar to current non-es|ql reports from discover UI, it doesn't use
saved searches but only relies on the state from the URL. This probably
will be improved in https://github.com/elastic/kibana/issues/151190 to
support both.
- Uses different CSV settings for functionality like checking formulas,
escapes, bom, max size, etc...
- Keeps regular CSV features like cancelation and giving event loop a
break (even though those are not needed for now for es|ql since the
number of results is limited)


Some notable differences compared to regular discover search / csv
reports:
- A lot simpler, as it doesn't use search source and field formats 
- No pagination, less CPU heavy as esql responses are limited to 10000
results and a single request

miltonhultgren committed:
[Obs AI Assistant] Regenerate from last user message (#174911)

## Summary

Changes the Regenerate functionality to look back to the last written
message from the user and use that as the cut of point for where to
generate the next messages.

maryam-saeidi committed:
[Alert details page][Custom threshold] Add group by and tags fields to the alert details page summary (#174254)

Closes #155698
Closes #174236

## Summary

This PR adds group and tag information to the alert details page summary
fields for the custom threshold rule:


![image](https://github.com/elastic/kibana/assets/12370520/d93e32bd-99ae-414f-87ed-5ca5884acfa6)


## 🧪 How to test
- Enable alert details page config
    ```
    xpack.observability.unsafe.alertDetails.observability.enabled: true
    ```
- Create a custom threshold rule with a group by and add more than 3
rule tags
- After the alert is fired, go to the alert details page (from alert
flyout or menu action in the alert table); you should see the group by
and tags information there.

Commit:
ci: move remaining Cypress tests behind GH label (#174523)

Move the remaining Security Solution Cypress tests behind the
GitHub flag `ci:all-cypress-suites`. This is a follow-up to PR #173815.

---------

Co-authored-by: Yngrid Coello <yngrid.coello@elastic.co>

juliaElastic committed:
[Fleet] validate all agent policies before updating them on output delete (#174921)

## Summary

Closes https://github.com/elastic/kibana/issues/165817

Moved the logic of validating agent policies before updating any agent
policies on output delete. This fixes the scenario where the output
delete fails on an agent policy with fleet-server integration, but some
agent policies were already updated by then.

To verify:
- create a logstash output and make it default for data and monitoring
output
- create a fleet server policy (with fleet server integration) with data
and monitoring output set to elasticsearch output
- create another agent policy with data and monitoring output set to
elasticsearch output
- attempt to delete the elasticsearch output, expect a UI error
- check that the agent policies are still referencing the elasticsearch
output, not the logstash one


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

SiddharthMantri committed:
Additional logging and add timeouts for flaky tests for Concurrent Session limit for users (#174748)

## Summary

Closes #149091 

This PR addresses the potential issue of a session not being found in
the session index by introducing a timeout before attempting to write
the next one. Passing these [changes through
FTR](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/4854)
make it pass 100% of the time with 400 test runs.

weltenwort committed:
[Fleet] Fix asset creation during custom integration installation (#174869)

dkirchan committed:
[Security] Security/Buildkite Analytics test suites integration (#174524)

## Summary

This PR introduces the integration of the following categories of tests
with Buildkite Test Suites.
- Security Solution Cypress - [Buildkite test
suite](https://buildkite.com/organizations/elastic/analytics/suites/ci-security-solution-cypress?branch=all+branches)
- Defend Workflows - [Buildkite test
suite](https://buildkite.com/organizations/elastic/analytics/suites/ci-defend-workflows?branch=all+branches)
- Threat Intelligence - [Buildkite test
suite](https://buildkite.com/organizations/elastic/analytics/suites/ci-threat-intelligence?branch=all+branches)
- Serverless Security Solution Cypress - [Buildkite test
suite](https://buildkite.com/organizations/elastic/analytics/suites/serverless-security-solution-cypress-1?branch=all+branches)

It also addresses a wrong package.json script invocation and the
parallelism that is used in the pipeline which was reduced for some
targets. Check File:
.buildkite/pipelines/security_solution/security_solution_cypress.yml


## Initial PR
https://github.com/elastic/kibana/pull/173005 - Status Closed

tonyghiani committed:
[Logs Explorer] Fix caching on data views updates (#174893)

## 📓 Summary

Closes #174538 

This work fixes an issue which prevented newly created data views from
being shown in the Log Explorer selector when performing client
navigations.

This issue was caused by a globally instantiated cache, which was not
recreated when the page was left and revisited later.

It now recreates correctly the initial context, including the cache, to
display the correct list of data views. The same fix applies to
integrations and datasets, which had their cache created in the same
way.


https://github.com/elastic/kibana/assets/34506779/4c53072a-7880-434c-9afc-d585a43610a2

Co-authored-by: Marco Antonio Ghiani <marcoantonio.ghiani@elastic.co>

tonyghiani committed:
[Logs Explorer] Update log detail flyout design (#174923)

## 📓 Summary

Update the log detail flyout to reflect some minor changes from the
[latest
design](https://www.figma.com/file/yEey1KMFFuwS4sbpErWIrw/Log-details-flyout?type=design&node-id=406-72192&mode=design&t=j588kwYms7Uibeoa-0):
- Merge Service and Infrastructure highlight sections
- Remove `event` token icon
- Update highlight fields design
- Put into a badge the `namespace` value
- Reorder fields


![localhost_5601_ftw_app_observability-log-explorer__pageState=(breakdownField_log
level,columns_!((field_service name,width_240),(field_host
name,width_320),(field_content)),controls_(data_stream
namespace_(mode_include,selection_(selectedOp](https://github.com/elastic/kibana/assets/34506779/c5dd5843-e87b-4bf7-a0a5-0c3302994ac4)

---------

Co-authored-by: Marco Antonio Ghiani <marcoantonio.ghiani@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Bamieh committed:
Integrate zh-CN translations from 8.12.0 extraction

Bamieh committed:
Integrate ja-JP translations from 8.12.0 extraction

Bamieh committed:
Integrate fr-FR translations from 8.12.0 extraction

Bamieh committed:
Merge branch 'i18n/sync_8.12.0_translations_to_main' of github.com:Bamieh/kibana into i18n/sync_8.12.0_translations_to_main
